controllers/medusa/medusaconfiguration_controller.go

```diff
@@ -74,15 +80,26 @@ func (r *MedusaConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	// Check if the referenced secret exists
 	if configuration.Spec.StorageProperties.StorageSecretRef.Name != "" {
-		err = r.CheckSecretPresence(ctx, r.Client, req, configuration.Spec.StorageProperties.StorageSecretRef.Name)
+		secret, err := r.GetSecret(ctx, r.Client, req, configuration.Spec.StorageProperties.StorageSecretRef.Name)
 		if err != nil {
 			logger.Error(err, "Failed to get MedusaConfiguration referenced secret")
 			configuration.Status.SetCondition(medusav1alpha1.ControlStatusSecretAvailable, metav1.ConditionFalse)
 			configuration.Status.SetConditionMessage(medusav1alpha1.ControlStatusSecretAvailable, err.Error())
 			r.patchStatus(ctx, configuration, patch, logger)
 			return ctrl.Result{}, err
-		} else {
-			configuration.Status.SetCondition(medusav1alpha1.ControlStatusSecretAvailable, metav1.ConditionTrue)
+		}
+		configuration.Status.SetCondition(medusav1alpha1.ControlStatusSecretAvailable, metav1.ConditionTrue)
+
+		patch := client.MergeFrom(secret.DeepCopy())
+		if secret.Labels == nil {
+			secret.Labels = make(map[string]string)
+		}
+		if secret.Labels[MedusaStorageSecretIdentifierLabel] != utils.HashNameNamespace(secret.Name, secret.Namespace) {
+			secret.Labels[MedusaStorageSecretIdentifierLabel] = utils.HashNameNamespace(secret.Name, secret.Namespace)
+			if err = r.Client.Patch(ctx, secret, patch); err != nil {
+				logger.Error(err, "Failed to patch Medusa Bucket Secret with required label")
+				return ctrl.Result{RequeueAfter: r.DefaultDelay}, err
```

Comment:
if err != nil, then RequeueAfter shouldn't be set. The err alone will do the correct requeue with a proper backoff strategy.